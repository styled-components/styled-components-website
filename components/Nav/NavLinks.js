import React from 'react'
import styled from 'styled-components'
import { I18n } from 'react-i18next'

import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import NavSeparator from './NavSeparator'
import Link from '../Link'

import {
  DEFAULT_TRANSLATION,
} from '../../constants/i18n'

import {
  addLanguageToPath,
} from '../../utils/translations'

const Wrapper = styled.nav`
  display: flex;
  align-items: center;
  flex: 0 0 auto;
`

const NavLink = styled(Link).attrs({
  unstyled: true
})`
  flex: 0 0 auto;
  display: inline-block;
  line-height: ${rem(navbarHeight)};
  transition: opacity 0.2s, transform 0.2s;
  cursor: pointer;

  letter-spacing: ${rem(0.4)};
  color: currentColor;

  &:hover,
  &:focus {
    opacity: 0.8;
  }

  &:active {
    transform: scale(0.95);
    opacity: 0.6;
  }
`

const NavLinks = () => (
  <I18n ns={DEFAULT_TRANSLATION}>
    {(translate, { i18n }) => (
      <Wrapper>
        <NavLink
          href="/"
          as={addLanguageToPath(i18n, '/')}
          >
            {translate('homepage')}
          </NavLink>
          <NavSeparator />
          <NavLink
            href="/docs"
            as={addLanguageToPath(i18n, '/docs')}
          >
            {translate('documentation')}
          </NavLink>
          <NavSeparator />
          <NavLink
            href="/ecosystem"
            as={addLanguageToPath(i18n, '/ecosystem')}
          >
            {translate('ecosystem')}
          </NavLink>
        </Wrapper>
    )}
  </I18n>
)

export default NavLinks
