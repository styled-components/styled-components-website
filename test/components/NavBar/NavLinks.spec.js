import React from 'react';
import renderer from 'react-test-renderer';
import NavLinks from '../../../components/Nav/NavLinks';

test('NavLinks renders correctly', () => {
  const tree = renderer.create(<NavLinks />).toJSON();

  expect(tree).toMatchSnapshot();
});
