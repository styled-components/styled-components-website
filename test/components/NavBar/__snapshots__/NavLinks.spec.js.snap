// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks renders correctly 1`] = `
.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 0.2777777777777778rem;
  height: 0.2777777777777778rem;
  margin: 0 0.8333333333333334rem;
  border-radius: 50%;
  background: currentColor;
  opacity: 0.35;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 1.6666666666666667rem;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
  -webkit-letter-spacing: 0.022222222222222223rem;
  -moz-letter-spacing: 0.022222222222222223rem;
  -ms-letter-spacing: 0.022222222222222223rem;
  letter-spacing: 0.022222222222222223rem;
  color: currentColor;
}

.c1:hover,
.c1:focus {
  opacity: 0.8;
}

.c1:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

<nav
  className="c0"
>
  <a
    className="c1"
    href="/docs"
  >
    Documentation
  </a>
  <span
    className="c2"
  />
  <a
    className="c1"
    href="/showcase"
  >
    Showcase
  </a>
  <span
    className="c2"
  />
  <a
    className="c1"
    href="/ecosystem"
  >
    Ecosystem
  </a>
  <span
    className="c2"
  />
  <a
    className="c1"
    href="/releases"
  >
    Releases
  </a>
</nav>
`;
